src/ui/App.tsx

```diff
@@ -102,14 +102,41 @@ function App(): JSX.Element {
   // network.
   const numCardsRendered = 10;
 
+  const [cardRotations, setCardRotations] = useState(new WeakMap<Thread, number>());
+  const [cardOffsets, setCardOffsets] = useState(new WeakMap<Thread, number>());
+
+  // Returns [rotation, offset]
+  function getCardPostionForThread(thread: Thread): number[] {
+    let rotation = cardRotations.get(thread);
+    if (!rotation) {
+      rotation = 2 * (Math.random() - 0.5);
+      setCardRotations(cardRotations.set(thread, rotation));
```

Comment:
I'm not sure how expensive hooks setters are, but I think they do a bunch of sync work. Could compute all the rotations up front and then call setCardRotations and setCardOffsets once instead of in a loop.